lib/services/service_brokers/v2/response_parser.rb

```diff
@@ -28,7 +28,7 @@ def parse_provision_or_bind(path, response)
           when 409
             FailingValidator.new(Errors::ServiceBrokerConflict)
           when 422
-            FailWhenValidator.new('error', ['AsyncRequired'], Errors::AsyncRequired,
+            FailWhenValidator.new('error', { 'AsyncRequired' => Errors::AsyncRequired, 'RequiresApp' => Errors::AppRequired },
```

Comment:
There's a very subtle mistake here. The RequiresApp error is only appropriate for the bind request, but this implementation would also allow provision requests to pass this error. If you write two tests, one for provision and one for bind, both of which pass the RequiresApp error, both tests will show the client raising AppRequired error, even though in the case of provision, the RequiresApp error should be ignored.

Look just a few lines above this at the IfElsePathMatcher. This allows us to give a condition and two validators; one runs if the condition is true, and the other validator runs if the condition is false. We can use this to make sure that we have two different FailWhenValdiators, one for provision and one for bind.